dev-tools/omdb/src/bin/omdb/mgs/dashboard.rs

```diff
@@ -0,0 +1,1115 @@
+        loop {
+            let values = sensor_data(&mut input, metadata).await?;
+
+            if values.time == 0 {
+                break;
+            }
+
+            if values.time != last + 1 {
+                dashboard.gap(values.time - last - 1);
+            }
+
+            last = values.time;
+            dashboard.values(&values);
+        }
+
+        update = false;
+    }
+
+    // setup terminal
+    enable_raw_mode()?;
+    let mut stdout = io::stdout();
+    execute!(stdout, EnterAlternateScreen, EnableMouseCapture)?;
+    let backend = CrosstermBackend::new(stdout);
+    let mut terminal = Terminal::new(backend)?;
+
+    let res = 'outer: loop {
+        match run_dashboard(&mut terminal, &mut dashboard, force) {
+            Err(err) => {
+                break Err(err);
+            }
+            Ok(true) => {
+                break Ok(());
+            }
+            _ => {}
+        }
+
+        force = false;
+        let now = secs()?;
+
+        if update && now != last {
+            let kicked = Instant::now();
+            let f = sensor_data(&mut input, metadata);
+            last = now;
+
+            while Instant::now().duration_since(kicked).as_millis() < 800 {
+                tokio::time::sleep(Duration::from_millis(10)).await;
+
+                match run_dashboard(&mut terminal, &mut dashboard, force) {
+                    Err(err) => {
+                        break 'outer Err(err);
+                    }
+                    Ok(true) => {
+                        break 'outer Ok(());
+                    }
+                    _ => {}
+                }
+            }
+
+            let values = f.await?;
+            dashboard.values(&values);
+            force = true;
+            continue;
+        }
+
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    };
+
+    // restore terminal
+    disable_raw_mode()?;
+    execute!(
+        terminal.backend_mut(),
+        LeaveAlternateScreen,
+        DisableMouseCapture
+    )?;
+    terminal.show_cursor()?;
+
+    if let Err(err) = res {
+        println!("{err:?}");
+    }
+
+    Ok(())
+}
+
+fn draw_graph(f: &mut Frame, parent: Rect, graph: &mut Graph, now: u64) {
+    //
+    // We want the right panel to be 31 characters wide (a left-justified 20
+    // and a right justified 8 + margins), but we don't want it to consume
+    // more than 80%; calculate accordingly.
+    //
+    let r = std::cmp::min((31 * 100) / parent.width, 80);
+
+    let chunks = Layout::default()
+        .direction(Direction::Horizontal)
+        .constraints(
+            [Constraint::Percentage(100 - r), Constraint::Percentage(r)]
+                .as_ref(),
+        )
+        .split(parent);
+
+    let latest = now as i64 - graph.offs as i64;
+    let earliest = Local.timestamp_opt(latest - graph.width as i64, 0).unwrap();
+    let latest = Local.timestamp_opt(latest, 0).unwrap();
+    let fmt = "%Y-%m-%d %H:%M:%S";
```

Comment:
I don't see anything wrong with this, but wondering: could we use [.to_rfc3339()](https://docs.rs/chrono/0.4.33/chrono/struct.DateTime.html#method.to_rfc3339) instead? Or maybe `to_rfc3339_opts` with `use_z: true`?

Reply:
It's a matter of horizontal screen real estate -- we don't get *quite* the exact format we want with `to_rfc3339_opts` in that it consumes maximum width for the TZ offset. I'll add a comment.